File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/other/JigsawPieceCurrentPositionResponse.java
package com.kafkaquest.kq.common.util.models.events.other;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class JigsawPieceCurrentPositionResponse {
    private UUID pieceId;
    private int rowPosition;
    private int columnPosition;
}

File: kq-projection-dotnet/kq-projection/Common/Models/Events/JigsawPieceCurrentPositionResponse.cs
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Events
{
    public class JigsawPieceCurrentPositionResponse
    {
        [JsonPropertyName("pieceId")]
        public Guid pieceId { get; set; }

        [JsonPropertyName("rowPosition")]
        public int rowPosition { get; set; }

        [JsonPropertyName("columnPosition")]
        public int columnPosition { get; set; }
    }
}
